src/variables/ShellCommandVariable_CaretPosition.ts

```diff
@@ -0,0 +1,98 @@
+import {getEditor} from "../Common";
+import {addShellCommandVariableInstructions} from "./ShellCommandVariableInstructions";
+import {IParameters, ShellCommandVariable} from "./ShellCommandVariable";
+import {IAutocompleteItem} from "../settings/setting_elements/Autocomplete";
+
+export class ShellCommandVariable_CaretPosition extends ShellCommandVariable {
+    static variable_name = "caret_position";
+    static help_text = "Gives the line number and column position of the current caret position as 'line:col'. Get only the line number using {{caret_position:line}}, and only the column with {{caret_position:column}}. Line and column numbers are 1-indexed.";
+
+    protected static readonly parameters: IParameters = {
+        mode: {
+            options: ["line", "column"],
+            required: false,
+        },
+    };
+
+    protected arguments: {
+        mode: string;
+    }
+
+    generateValue(): string {
+        // Check that we are able to get an editor
+        let editor = getEditor(this.app);
+        if (null === editor) {
+            // Nope.
+            this.newErrorMessage("Could not get an editor instance! Please raise an issue in GitHub.");
+            return null;
+        }
+
+        let position = editor.getCursor('to');
+        const line = position.line + 1; // editor position is zero-indexed, line numbers are 1-indexed
+        const col = position.ch + 1; // editor position is zero-indexed, column positions are 1-indexed
+        
+        if (Object.keys(this.arguments).length > 0) {
+            switch (this.arguments.mode.toLowerCase()) {
+                case "line":
+                    return `${line}`;
+                case "column":
+                    return `${col}`;
+                default:
+                    Error("ShellCommandVariable_CaretPosition: Unrecognised argument: "+this.arguments.mode);
```

Comment:
Variables have their own error handler.
```suggestion
                    this.newErrorMessage("Unrecognised argument: "+this.arguments.mode);
```

This shows the error message to a user in various ways: if executing a command, it will be shown in a notification balloon; if writing commands in settings, the preview text will show the error message. Variable name is automatically added to the message.